### PR TITLE
Feature/backbone modal modal

### PR DESCRIPTION
**Pull request by marsh73**

### CODE REVIEW

File: scarlet/cms/static/scarlet/source/js/helpers/Modal.js [hunk 1]
```diff
@@ -97,41 +129,74 @@ class Modal {
...
+			return Math.max(a, getModalContent(b).width())
+		}, 0)
+}
+
+/**
+ * helper to find doc body content for modal
+ * @param  Object - iframe wrapper
+ * @return Object - #content node inside iframe
+ */
+function getModalContent (wrap) {
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/js/helpers/Modal.js]: would it make sense to name the param iframe?

File: scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss [hunk 2]
```diff
@@ -1,6 +1,7 @@
   @import '~jquery-ui/themes/ui-darkness/jquery-ui.css';
+  @import '../mixins/mixins';
 
-.ui-dialog-content {
+.frame-wrapper {
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss]: is this the outermost parent element?

File: scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss [hunk 3]
```diff
@@ -38,8 +39,19 @@
     padding: 0;
   }
 }
+.dialog__no-title {
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss]: what's the difference between .dialog__ and .modal__ (below)?  the layout flow would be easier to follow if .modal was the namespace for the entire layout (.modal being the outermost block)

File: scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss [hunk 4]
```diff
@@ -38,8 +39,19 @@
...
+.modal__frameWrap{
+  height: 100%;
+  width: 100%;
+  // STACKED
+  position: absolute;
+  top: 0;
+}
 
-body.modalDialog__body{
+body.modal__body, {
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss]: is tag qualification necessary?

File: scarlet/cms/static/scarlet/source/js/helpers/Modal.js [hunk 5]
```diff
@@ -97,41 +129,74 @@ class Modal {
...
+	}
 }
 
 /**
  * For Click to open window 
  * @param  {object} event
  * @param  {Function} callback
  */
-const clickOpenModal = function (e, cb) {
+const clickOpenModal = function (e, name, cb, tags) {
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/js/helpers/Modal.js]: maybe name this simply openModal instead, so the event type isn't coupled to the method name

File: scarlet/cms/static/scarlet/source/js/helpers/Modal.js [hunk 6]
```diff
@@ -97,41 +129,74 @@ class Modal {
...
+		return $('<div class="modal__frameWrap modal__inline"></div>')
+	 		.html('<iframe id="'+name+'" src="'+url+'" style="border: 0px; " src="' + url + '" width="100%" height="100%"></iframe>')
+	} else {
+
+		return $('<div class="modal__framesContainer"></div>')
+			.html('<div class="modal__frameWrap"><iframe id="'+name+'" style="border: 0px; " src="' + url + '" width="100%" height="100%"></iframe></div>')
+			.dialog(options)
+	}
+}
+
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/js/helpers/Modal.js]: DRY -  a method that returns the modal content template would help

File: scarlet/cms/static/scarlet/source/js/helpers/Modal.js [hunk 7]
```diff
@@ -19,46 +19,67 @@ class Modal {
...
+			this.onLoad(qry, frame, tags)
 		})
 
 	}
 
 	/**
+	 * fires on iFrame Load
+	 * @param  {object}
+	 */
+	onLoad (qry, frame, tags) {
```
Comment (annaocd) [inline scarlet/cms/static/scarlet/source/js/helpers/Modal.js]: suggestion - avoid abbreviated parameters

**Comment by annaocd** [comment]
👍  fine to merge but the html structures and naming conventions need polish

File: scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss (hunk 3 above)
Comment (marsh73) [inline scarlet/cms/static/scarlet/source/stylesheets/views/modal.scss]: dialog is the jquery-ui component... modal is our component

**Pull request merged**